src/components/smartSearch/filters/SurveyOption/index.tsx

```diff
@@ -0,0 +1,231 @@
+    onCancel: () => void;
+}
+
+interface InternalConfig {
+    survey?: number;
+    question?: number;
+    operator: CONDITION_OPERATOR;
+    options: number[];
+}
+
+const SurveyOption = (
+    { onSubmit, onCancel, filter: initialFilter }: SurveyOptionProps,
+): JSX.Element => {
+    const { orgId } = useRouter().query;
+    const surveysQuery = useQuery(['surveysWithElements', orgId], getSurveysWithElements(orgId as string));
+    const surveys = surveysQuery.data || [];
+
+    const { filter, setConfig, setOp } = useSmartSearchFilter<InternalConfig>(initialFilter, {
+        operator: CONDITION_OPERATOR.ALL,
+        options: [],
+    });
+
+    useEffect(() => {
+        if (surveys.length) {
+            setConfig({
+                ...filter.config,
+                question: filter.config.question ||
+                surveys[0].elements
+                    .find(e => e.type === ELEMENT_TYPE.QUESTION &&
+                    e.question.response_type === RESPONSE_TYPE.OPTIONS)?.id,
+                survey: filter.config.survey || surveys[0].id,
+            });
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [surveys]);
+
+    // check if there are questions with response type of 'text'
+    const validQuestions = surveys
+        .find(s => s.id === filter.config.survey)?.elements
+        .filter(e => e.type === ELEMENT_TYPE.QUESTION &&
+            e.question.response_type === RESPONSE_TYPE.OPTIONS) || [];
+
+    const validOptions = surveys
+        .find(s => s.id === filter.config.survey)?.elements
+        .find(e => e.id === filter.config.question)?.question.options || [];
+
+    // convert filter.config.options to an array of Zetkin option objects keeping the correct order
+    const selectedOptions = filter.config.options.map(option => validOptions.find(o => o?.id === option))
+        .filter(o => o) as ZetkinOption[];
+
+    //submit if there is valid survey, valid questions and at least one option chosen
+    const submittable = filter.config.survey && filter.config.question && filter.config.options?.length;
+
+    //event handlers
+    const handleSubmit = (e: FormEvent) => {
+        e.preventDefault();
+        onSubmit(filter as ZetkinSmartSearchFilter<SurveyOptionFilterConfig>);
+    };
+
+    const handleQuestionSelectChange = (questionValue: string) => {
+        if (questionValue === DEFAULT_VALUE) {
+            setConfig({ ...filter.config, question: undefined });
+        }
+        else {
+            setConfig({ ...filter.config, question: +questionValue });
+        }
+    };
+
+    const handleSurveySelectChange = (surveyValue: string) => {
+        const newQuestion = surveys.find(s => s.id === +surveyValue)?.elements
+            .find(e => e.type === ELEMENT_TYPE.QUESTION &&
+            e.question.response_type === RESPONSE_TYPE.OPTIONS);
+        setConfig({ ...filter.config, question: newQuestion?.id, survey: +surveyValue });
```

Comment:
Is there a bug here where the options aren't reset when the question is? I'm guessing this could result in a bug when doing this:

1. Pick a question 1
2. Pick a few options A and B
3. Pick another question 2 (options goes blank)
4. Save

Would the config then not become question 2, options A+B (even though they belong to question 1)?

Reply:
yep, fixing